crm_lead_product/report/crm_product_report.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (C) 2017-19 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from psycopg2.extensions import AsIs
+
+from odoo import fields, models, tools
+
+
+class ActivityReport(models.Model):
+    """CRM Lead Analysis"""
+
+    _name = "crm.product.report"
+    _auto = False
+    _description = "CRM Pipeline by Product Analysis"
+    _rec_name = "id"
+
+    active = fields.Boolean(readonly=True)
```

Comment:
Suggestion to have Lead/Opportunity as a related field and use the lead line as the base model.
```suggestion
    active = fields.Boolean(readonly=True)
    lead_id = fields.Many2one("crm.lead", "Lead/Opportunity", readonly=True)
```